### DISCUSSION TITLE
Error: "Something went wrong on our end. Please try again."

### DISCUSSION THREAD

**Discussion by nl-danish** [hyochan/react-native-iap#2798] [Q&A]
Hi Team,

We were currently working on switching our current app's plan with upgrade and downgrade feature,
However after making all the required changes to the RNiap package, we seem to have this error from the the Play Console, and if we try to comment out the payment token, everything works fine (Works as we are buying single plan perfectly)

Can anybody help us with what's going wrong? As, we're not able to see any logs that points out to any of the code failing.

Expected Behavior:
The expected behaviour is that the user would get a modal from google play console, that it's ready to upgrade the existing plan to the next one with the prorated amount.

Environment:
react-native-iap: "12.0.0"
react-native: ^0.66.5
Platforms (iOS, Android, emulator, simulator, device): Android

Screenshots:
![image](https://github.com/dooboolab-community/react-native-iap/assets/130751561/7fc1d346-b074-4c6a-9789-a6ff1cea4fe4)

**Comment by Nikhil-MathBud** [comment]
Hi @nl-danish  were you able to fix problem, I am facing  similar issue, 
just that in my case, ReplacementMode.CHARGE_PRORATED_PRICE is not working, giving same message,  but ReplacementMode.WITHOUT_PRORATION works fine. 
https://github.com/hyochan/react-native-iap/issues/2940 
Note- Purchase token is required for upgrading the plan, but even with that its throwing error.
